pages/api/bib/index.ts

```diff
@@ -0,0 +1,72 @@
+import type { BibParams, BibResponse } from "../../../src/types/bibTypes"
+import {
+  getBibQuery,
+  isNyplBibID,
+  standardizeBibId,
+} from "../../../src/utils/bibUtils"
+import nyplApiClient from "../../../src/server/nyplApiClient"
+import {
+  DISCOVERY_API_NAME,
+  DISCOVERY_API_SEARCH_ROUTE,
+} from "../../../src/config/constants"
+import { appConfig } from "../../../src/config/config"
+
+export async function fetchBib(
```

Comment:
Because of the file directory and its location, one would think that `/research/research-catalog/api/bib` is a valid API route, but it's not. This is the location for the function used in the `.../api/bib/:id` route and in the Bib page's `getServerSideProps` and it makes sense and it's great to reuse this function in both places.

So I think this would be better in a different directory, perhaps `src/server`? Just an idea, let me know what you think. Otherwise, this file and function is fine. It does need tests, however, which can be covered in this PR or another. Mocking `nyplApiClient` can be difficult but this was my attempt to mocking it: https://github.com/NYPL/research-catalog/pull/65/files#diff-713db81b94eaf0bbf2724f5aa23fbd2139a9ba3c70fa11bb76fcf661b3dbdbf7R3

Reply:
This is an interesting observation. My initial thought was to keep the fetching functions as close as possible to the route handlers, but they technically don't need to be there. Happy to move this as well as the other fetchers to the server directory.

Reply:
Does there have to be an index page in this directory? Is there any use case for a /api/bib route without a bib id?

Reply:
@charmingduchess No there isn't. As soon as we move the fetcher functions to the /server directory (as part of a follow-up ticket) we will only need the [id] file.